pkg/collector/corechecks/snmp/internal/session/session.go

```diff
@@ -86,6 +87,8 @@ func NewGosnmpSession(config *checkconfig.CheckConfig) (Session, error) {
 		}
 		s.gosnmpInst.Community = config.CommunityString
 	} else if config.User != "" {
+		snmp.CheckAndSetDefaultProtocols(config.AuthKey, &config.AuthProtocol, config.PrivKey, &config.PrivProtocol)
```

Comment:
Similar to other cases above, IMHO we should avoid passing pointers.

I have a proposal for this case, why not change `gosnmplib.GetAuthProtocol` and `gosnmplib.GetPrivProtocol` to accept both the key and proto.

```
		authProtocol, err := gosnmplib.GetAuthProtocol(config.AuthKey, config.AuthProtocol)
		if err != nil {
			return nil, err
		}
```

And then handle the setting the default in `GetAuthProtocol/GetPrivProtocol` directly?